notebooks/11_temporal_probability_models/index.md

```diff
@@ -0,0 +1,603 @@
+&= \alpha P(e_{t+1} |X_{t+1}) P(X_{t+1} | e_{1:t}) \quad \text{(by the sensor Markov assumption).}
+\end{align*}
+$$
+
+Here $\alpha$ is a normalizing constant used to make probabilities sum up to 1. The second term, $P(X_{t+1} | e_{1:t})$ represents a one-step prediction of the next state, and the first term updates this with the new evidence; notice that $P(e_{t+1} |X_{t+1})$ is obtainable directly from the sensor model.
+Now we obtain the one-step prediction for the next state by conditioning on the current state $X_t$:
+
+$$
+\begin{align*}
+P(X_{t+1} | e_{1:t+1}) &= \alpha P(e_{t+1} |X_{t+1}) \sum_{x_t} P(X_{t+1} | x_t, e_{1:t})P(x_t | e_{1:t})  \\
+&= \alpha P(e_{t+1} |X_{t+1}) \sum_{x_t} P(X_{t+1} | x_t)P(x_t | e_{1:t}) \quad \text{(Markov assumption).}
+\end{align*}
+$$
+
+Within the summation, the first factor comes from the transition model and the second comes from the current state distribution. Hence, we have the desired recursive formulation. We can think of the filtered estimate $P(X_t | e_{1:t})$ as a "message" $f_{1:t}$ that is propagated forward along the sequence, modified by each transition and updated by each new observation. The process is given by
+
+$$
+f_{1:t+1} = \alpha \text{FORWARD}(f_{1:t}, e_{t+1}) ,
+$$
+
+where FORWARD implements the update described in previous equation and the process begins with $f_{1:0} = P(X_0)$. When all the state variables are discrete, the time for each update is constant (i.e., independent of t), and the space required is also constant.
+
+### An Example
+Let us illustrate the filtering process for two steps in the basic umbrella example. That is, we will compute $P(R_2 | u_{1:2})$ as follows:
+
+- On day 0, we have no observations, only the security guard’s prior beliefs; let’s assume that consists of $P(R_0) = <0.5, 0.5>$.
+- On day 1, the umbrella appears, so $U_1 =true$. The prediction from $t=0$ to $t=1$ is
+
+$$
+\begin{align*}
+P(R_1) &= \sum_{r_0} P(R_1 | r_0)P(r_0) \\
+& = <0.7, 0.3> \times 0.5 + <0.3, 0.7> \times 0.5 = <0.5, 0.5> .
+\end{align*}
+$$
+
+Then the update step simply multiplies by the probability of the evidence for $t=1$ and normalizes:
+
+$$
+\begin{align*}
+P(R_1 | u_1) &= \alpha P(u_1 |R_1)P(R_1) = \alpha <0.9, 0.2><0.5, 0.5> \\
+& = \alpha <0.45, 0.1> \approx <0.818, 0.182> .
+\end{align*}
+$$
+
+- On day 2, the umbrella appears, so $U_2 =true$. The prediction from $t=1$ to $t=2$ is
+
+$$
+\begin{align*}
+P(R_2 | u_1) &= \sum_{r_1} P(R_2 | r_1)P(r_1 | u_1) \\
+& = <0.7, 0.3> \times 0.818 + <0.3, 0.7> \times 0.182 \approx <0.627, 0.373>,
+\end{align*}
+$$
+
+and updating it with the evidence for $t=2$ gives
+
+$$
+\begin{align*}
+P(R_2 | u_1, u_2) &= \alpha P(u_2 |R_2)P(R_2 | u_1) = \alpha <0.9, 0.2><0.627, 0.373> \\
+& = \alpha <0.565, 0.075> \approx <0.883, 0.117> .
+\end{align*}
+$$
+
+Intuitively, the probability of rain increases from day 1 to day 2 because rain persists.
+
+## Prediction
```

Comment:
try to make different sections of your lecture note more relevant to one another; don't jump from a topic to another without proper preparation